src/logging.cc

```diff
@@ -2425,7 +2426,7 @@ void GetExistingTempDirectories(vector<string>* list) {
 }
 
 void TruncateLogFile(const char *path, uint64 limit, uint64 keep) {
-#ifdef HAVE_UNISTD_H
+#if defined(HAVE_UNISTD_H) || defined(GLOG_OS_WINDOWS)
```

Comment:
I suggest integrating the detection of `_chsize_s` into `CMakeLists.txt` and using the corresponding define instead of relying on ad-hoc preprocessor logic. This way, it is easier to see whether the function is available and can be used already when configuring the project.

Something along the lines should do the trick:
```cmake
check_cxx_symbol_exists (_chsize_s io.h HAVE__CHSIZE_S)
```
For Bazel the `HAVE__CHSIZE_S` define can be added here:

https://github.com/google/glog/blob/857df01007020bc4767d9f819891fb1b2da7cb86/bazel/glog.bzl#L112-L118

Reply:
hi @sergiud , I just added these code as your suggestion